### PR TITLE
Disable running figure tests by default

### PR DESCRIPTION
**Pull request by Cadair**
This changes the defaults for `sunpy.self_test` and `python setup.py test` to not run the tests with the `pytest.mark.figure` mark.

It also adds a keyword argument to `self_test` and a `--figure` flag to `setup.py test` to re-enable these tests.

### CODE REVIEW

File: astropy_helpers [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 160b48587680e11f12b66a51641a2601b869d1fa
+Subproject commit e87287ac3c82159a6b8a578491f209b01c47e64a
```
Comment (dpshelio) [inline astropy_helpers]: Is this also updating astropy helpers?

**Comment by dpshelio** [comment]
:+1:  so we can merge my #1629 - but the astropy-helpers update is different than the one from #1627

File: astropy_helpers (hunk 1 above)
Comment (Cadair) [inline astropy_helpers]: yes to the next bug fix release, not 1.1

**Comment by Cadair** [comment]
Ok I am going to merge this, despite it only being a part -fix.

**Pull request merged**